File: Part2-Regression/Section 6 - Simple Linear Regression/R/simple_linear_regression.R
# Simple Linear Regression

# Importing the dataset
dataset = read.csv('Salary_Data.csv')

#splitting dataset into training and test set
# install.packages('caTools')

library(caTools)

set.seed(123)

split = sample.split(dataset$Salary,SplitRatio = 2/3)#name of dependent variable is salary
training_set = subset(dataset, split == TRUE)
test_set = subset(dataset,split == FALSE)

#fitting simple linear regression into the training set
regressor = lm(formula = Salary ~ YearsExperience,
               data = training_set)

#predicting the test set result

y_pred = predict(regressor, newdata = test_set)

#visualising the training set results
# install.packages('ggplot2')
ggplot() + 
  geom_point(aes(x = training_set$YearsExperience, y = training_set$Salary),
             colour = 'red') + #it is used to scatter plot
                              #it shows real value.That is original one
  geom_line(aes(x = training_set$YearsExperience, y = predict(regressor,newdata = training_set)),
            color = 'blue') + #it'll show predicted salaries of training set
  
  ggtitle('salary vs Experience(Training set)') + 
  xlab('years of Experience')+
  ylab('salary')


#visualising test set results

ggplot() + 
  geom_point(aes(x = test_set$YearsExperience,y = test_set$Salary),
             color = 'green')+
  geom_line(aes(x = training_set$YearsExperience,y = predict(regressor,newdata = training_set)),
            color = 'blue')+ 
  ggtitle('salary vs Experience(Test set)')+
  xlab('years of Experience')+
  ylab('salary')
  


  

File: Part2-Regression/Section 6 - Simple Linear Regression/Python/simple_linear_regression.py
# Simple Linear Regression

# Importing the libraries
import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

#importing the dataset
dataset = pd.read_csv('Salary_Data.csv')
X = dataset.iloc[:,:-1].values #independent vriables
y = dataset.iloc[:,1].values #dependent variables

#on looking output we find X is matrix as we have 1 column whereas in y we have no column therefore it is vector

#splitting the dataset into Training and Test set
from sklearn.model_selection import train_test_split
X_train,X_test, y_train, y_test =  train_test_split(X, y, test_size = 1/3, random_state = 0)

#X_train is matrix of indepenndent variable and y_train is dependent variable factor for trainig set
#X_test is matrix of indepenndent variable and y_test is dependent variable factor for test set

#fitting Simple Linear Regression into the training set